core/test/Cardano/Address/ScriptSpec.hs

```diff
@@ -276,6 +276,52 @@ spec = do
     describe "can perform roundtrip JSON serialization & deserialization" $
         it "fromJSON . toJSON === pure" $ property prop_jsonRoundtrip
 
+    describe "some JSON parsing error" $ do
+        it "Empty list" $ do
+            let err = "Error in $: The list inside a script is empty."
+            Json.eitherDecode @Script "{ \"any\": [] }"
+                `shouldBe` Left err
+
+        it "Empty list nested" $ do
+            let err = "Error in $.any[0]: The list inside a script is empty."
+            Json.eitherDecode @Script "{ \"any\": [ { \"all\": [] } ] }"
+                `shouldBe` Left err
+
+        it "Invalid type" $ do
+            let err = "Error in $.all[0].any[0]: expected Object or String, but encountered Number"
+            Json.eitherDecode @Script "{ \"all\": [ { \"any\": [1,2,3] } ] }"
+                `shouldBe` Left err
+
+        it "List too small" $ do
+            let err = "Error in $: At least must not be larger than the list of keys."
+            Json.eitherDecode @Script
+                "{ \"some\":\
+                \   { \"at_least\": 2\
+                \   , \"from\":\
+                \       [\"script_vkh18srsxr3khll7vl3w9mqfu55n6wzxxlxj7qzr2mhnyreluzt36ms\"\
+                \       ]\
+                \   }\
+                \}" `shouldBe` Left err
```

Comment:
What is the valid input as a script here? readme says:
```
all [script_vkh15thv7uqqd0jh48fv4achd7ufqu9kv98c78r40q7fpwylw2c6wft,script_vkh18w06sur9ll7pcw8nzzr9mllgzyrrraaqse8336k2ej2cyph0u2a]
```
which is not a JSON?
As far as I see JSON is not supported?
```
{ "some": { "from": [ "script_vkh1yf07000d4ml3ywd3d439kmwp07xzgv6p35cwx8h605jfx0dtd4a", "script_vkh1mwlngj4fcwegw53tdmyemfupen2758xwvudmcz9ap8cnqk7jmh4", "script_vkh1qw4l62k4203dllrk3dk3sfjpnh3gufhtrtm4qvtrvn4xjp5x5rt" ], "at_least": 2 } }
```
:point_down: 

```
Parsing of the script failed. The script should be composed of nested lists, and the verification keys should be either encoded as bech32.
```

Reply:
>  As far as I see JSON is not supported?

Indeed, the command uses a custom DSL which is not JSON (but close). Arguably, we could support both. These particular tests are testing the JSON instance however. There are tests for the script DSL in https://github.com/input-output-hk/cardano-addresses/blob/master/core/test/Cardano/Address/Script/ParserSpec.hs#L40-L52